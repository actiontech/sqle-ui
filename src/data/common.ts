export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}