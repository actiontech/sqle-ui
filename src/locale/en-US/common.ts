// eslint-disable-next-line import/no-anonymous-default-export
export default {
  username: 'username',
  password: 'password',

  unknownError: 'Unknown Error',

  ok: 'Ok',
  cancel: 'Cancel',
  submit: 'Submit',
  close: 'Close',
  edit: 'Edit',
  delete: 'Delete',
  reset: 'Reset',
  search: 'Search',
  retry: 'Retry',
  back: 'Back',
  upload: 'Upload',
  resetAndClose: 'Reset Form And Close',
  operateSuccess: 'Operate successfully',
  prevStep: 'Prev',
  nextStep: 'Next',

  expansion: 'Expand',
  collapse: 'Collapse',

  theme: {
    light: 'Light Theme',
    dark: 'Dark Theme',
  },

  logout: 'Logout',
  account: 'Account',

  nav: {
    title: 'SQLe',
  },

  request: {
    noticeFailTitle: 'Request Failed',
  },

  form: {
    placeholder: {
      input: 'Please input {{name}}',
      select: 'Please select {{name}}',
      searchInput: 'Input the {{name}} to search',
      searchSelect: 'Select the {{name}} to search',
    },
    rule: {
      require: 'Please input {{name}}',
      passwordNotMatch: 'The two passwords that you entered do not match!',
      email: 'The input is not valid E-mail!',
    },
  },
};
