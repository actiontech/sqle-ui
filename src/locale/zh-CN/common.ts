// eslint-disable-next-line import/no-anonymous-default-export
export default {
  username: '用户名',
  password: '密码',

  unknownError: '未知错误...',
  unknownStatus: '未知状态...',

  ok: '确认',
  cancel: '取消',
  submit: '提交',
  close: '关闭',
  edit: '编辑',
  modify: '修改',
  delete: '删除',
  reset: '重置',
  resetAll: '重置所有内容',
  search: '搜索',
  retry: '重试',
  back: '返回',
  more: '更多',
  upload: '选择文件上传',
  resetAndClose: '重置表单并关闭',
  operateSuccess: '操作成功',
  operate: '操作',

  prevStep: '上一步',
  nextStep: '下一步',

  expansion: '展开',
  collapse: '收起',

  showAll: '查看所有',
  showDetail: '查看详情',

  in: '在',
  on: '的',
  and: '且',
  at: '中',
  preview: '预览',

  theme: {
    light: '明亮模式',
    dark: '暗黑模式',
  },

  logout: '退出登录',
  account: '个人中心',

  nav: {
    title: 'SQLE',
  },

  request: {
    noticeFailTitle: '请求错误',
  },

  time: {
    hour: '小时',
    year: '年',
    month: '月',
    day: '天',
    no: '日',
    week: '星期',
    minute: '分钟',
    per: '每',
  },

  form: {
    placeholder: {
      input: '请输入{{name}}',
      select: '请选择{{name}}',
      searchInput: '请输入要搜索的 {{name}}',
      searchSelect: '请选择要搜索的 {{name}}',
    },
    rule: {
      require: '必须填写{{name}}',
      selectFile: '必须选择一个文件',
      passwordNotMatch: '两次密码输入不一致',
      email: '请输入有效的邮箱地址',
      startWithLetter: '必须要以字母开头',
      onlyLetterAndNumber: '只能包含字母、数字、中划线和下划线',
    },
  },

  cron: {
    mode: {
      select: '可视化选择',
      manual: '手工填写',
    },
    errorMessage: {
      invalid: '无效的cron表达式',
      mustBeString: 'cron表达式必须是一个字符串',
      mustBeArray: '变更的值必须是一个数组',
      lenMustBeFive: 'cron表达式必须只包含（分钟 小时 日期 月份 星期）5个元素',
      onlyHaveValidChar: 'cron表达式只能包含数字中划线(-),斜线(/),和逗号(,)',
      limit:
        '您的表达式中包含不合法的数值范围， minute(0-59), hour(0-23), day(1,31), month(1-12), week(0-6)',
    },
  },
};
