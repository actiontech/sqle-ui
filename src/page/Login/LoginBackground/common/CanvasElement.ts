import { IPosition } from './index.type';

abstract class CanvasElement {
  abstract position: IPosition;
  abstract canvas: HTMLCanvasElement;
  abstract context: CanvasRenderingContext2D;
  abstract draw(): void;
  abstract update(reachPosition: IPosition): void;
}

export default CanvasElement;
