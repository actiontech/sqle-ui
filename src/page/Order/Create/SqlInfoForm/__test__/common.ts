import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
