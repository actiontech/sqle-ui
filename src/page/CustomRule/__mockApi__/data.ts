import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
