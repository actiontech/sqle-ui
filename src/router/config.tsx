import React from 'react';
import { RouteProps } from 'react-router-dom';
import { RouterItem } from '../types/router.type';
import {
  PieChartOutlined,
  DesktopOutlined,
  ContainerOutlined,
  UserOutlined,
  DatabaseOutlined,
  AuditOutlined,
  ConsoleSqlOutlined,
  BarsOutlined,
  ProfileOutlined,
  SettingOutlined,
  NodeIndexOutlined,
  CiCircleOutlined,
} from '@ant-design/icons';
import { SystemRole } from '../data/common';

const Home = React.lazy(
  () => import(/* webpackChunkName: "Home" */ '../page/Home')
);

const Login = React.lazy(
  () => import(/* webpackChunkName: "Login" */ '../page/Login')
);

const User = React.lazy(
  () => import(/* webpackChunkName: "User" */ '../page/User')
);

const Rule = React.lazy(
  () => import(/* webpackChunkName: "Rule" */ '../page/Rule')
);

const DataSource = React.lazy(
  () => import(/* webpackChunkName: "DataSource" */ '../page/DataSource')
);

const RuleTemplate = React.lazy(
  () => import(/* webpackChunkName: "RuleTemplate" */ '../page/RuleTemplate')
);

const Account = React.lazy(
  () => import(/* webpackChunkName: "Account" */ '../page/Account')
);

const CreateOrder = React.lazy(
  () => import(/* webpackChunkName: "CreateOrder" */ '../page/Order/Create')
);

const OrderDetail = React.lazy(
  () => import(/* webpackChunkName: "OrderDetail" */ '../page/Order/Detail')
);

const OrderList = React.lazy(
  () => import(/* webpackChunkName: "Order" */ '../page/Order/List')
);

const Whitelist = React.lazy(
  () => import(/* webpackChunkName: "Whitelist" */ '../page/Whitelist')
);

const System = React.lazy(
  () => import(/* webpackChunkName: "System" */ '../page/System')
);

const WorkflowTemplate = React.lazy(
  () =>
    import(
      /* webpackChunkName: "WorkflowTemplate" */ '../page/WorkflowTemplate'
    )
);

const AuditPlan = React.lazy(
  () => import(/* webpackChunkName: "AuditPlan" */ '../page/AuditPlan')
);

const AuditPlanDetail = React.lazy(
  () =>
    import(/* webpackChunkName: "PlanDetail" */ '../page/AuditPlan/PlanDetail')
);

export const unAuthRouter: Array<RouteProps & { key: string }> = [
  {
    path: '/login',
    component: Login,
    exact: true,
    key: 'login',
  },
];

export const routerConfig: RouterItem[] = [
  {
    path: '/',
    exact: true,
    label: 'menu.dashboard',
    component: Home,
    icon: <PieChartOutlined />,
    key: 'dashboard',
  },
  {
    path: '/rule',
    exact: true,
    label: 'menu.rule',
    component: Rule,
    icon: <DesktopOutlined />,
    key: 'rule',
  },
  {
    path: '/account',
    exact: true,
    label: 'common.account',
    component: Account,
    hideInSliderMenu: true,
    key: 'account',
  },
  {
    label: 'menu.order',
    key: 'order',
    icon: <ConsoleSqlOutlined />,
    components: [
      {
        path: '/order',
        exact: true,
        label: 'menu.orderList',
        icon: <BarsOutlined />,
        component: OrderList,
        key: 'orderList',
      },
      {
        path: '/order/create',
        exact: true,
        label: 'menu.workflow',
        component: CreateOrder,
        icon: <DesktopOutlined />,
        key: 'orderCreate',
        hideInSliderMenu: true,
      },
      {
        path: '/order/:orderId',
        exact: true,
        label: 'menu.orderDetail',
        hideInSliderMenu: true,
        component: OrderDetail,
        key: 'orderDetail',
      },
    ],
  },
  {
    label: 'menu.platformManage',
    role: [SystemRole.admin],
    key: 'platformManage',
    icon: <ContainerOutlined />,
    components: [
      {
        path: '/user',
        exact: true,
        label: 'menu.user',
        component: User,
        icon: <UserOutlined />,
        key: 'user',
      },
      {
        path: '/data',
        key: 'data',
        label: 'menu.dataSource',
        icon: <DatabaseOutlined />,
        component: DataSource,
      },
      {
        path: '/rule/template',
        key: 'ruleTemplate',
        label: 'menu.ruleTemplate',
        icon: <AuditOutlined />,
        component: RuleTemplate,
      },
      {
        path: '/whitelist',
        key: 'Whitelist',
        label: 'menu.whitelist',
        component: Whitelist,
        icon: <ProfileOutlined />,
      },
      {
        path: '/system',
        key: 'System',
        label: 'menu.systemSetting',
        exact: true,
        component: System,
        icon: <SettingOutlined />,
      },
      {
        path: '/progress',
        key: 'progress',
        label: 'menu.progressManage',
        icon: <NodeIndexOutlined />,
        component: WorkflowTemplate,
      },
      {
        path: '/auditPlan/detail/:auditPlanName',
        key: 'auditPlanDetail',
        label: 'menu.auditPlane',
        hideInSliderMenu: true,
        component: AuditPlanDetail,
      },
      {
        path: '/auditPlan',
        key: 'auditPlan',
        label: 'menu.auditPlane',
        icon: <CiCircleOutlined />,
        component: AuditPlan,
      },
    ],
  },
];
