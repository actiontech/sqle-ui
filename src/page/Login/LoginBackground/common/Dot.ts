import { random } from '../../../../utils/Math';
import CanvasElement from './CanvasElement';
import { IPosition } from './index.type';

class Dot extends CanvasElement {
  public canvas: HTMLCanvasElement;
  public context: CanvasRenderingContext2D;
  public position: IPosition = { x: 0, y: 0 };
  private size: number;
  private color: string;
  private slope!: number;
  private offset!: number;
  private speed: number;
  private randomNum: number;
  private autoMove: boolean = true;

  constructor(
    canvas: HTMLCanvasElement,
    speed?: number,
    size?: number,
    color?: string,
    slope?: number,
    offset?: number
  ) {
    super();
    this.canvas = canvas;
    this.context = canvas.getContext('2d') as CanvasRenderingContext2D;
    this.size = size ?? 3;
    this.color = color ?? '#fff';
    this.speed = speed ?? 1;
    this.randomNum = Math.random() * 5 + 5;
    this.initPosition();
    if (!slope || !offset) {
      this.randomTrajectory();
    }
  }

  public draw(): void {
    this.context.beginPath();
    this.context.fillStyle = this.color;
    this.context.arc(
      this.position.x,
      this.position.y,
      this.size,
      0,
      Math.PI * 2
    );
    this.context.fill();
    this.context.closePath();
  }

  public update(reachPosition?: IPosition): void {
    let { x, y } = this.position;
    if (!reachPosition) {
      if (!this.autoMove) {
        return;
      }
      x += this.speed;
      if (x <= 0 || x >= this.canvas.width) {
        x = 0;
      }
      y = this.slope * x + this.offset;
    } else {
      x = x + ((reachPosition.x - x) * this.randomNum) / 50;
      y = y + ((reachPosition.y - y) * this.randomNum) / 50;
    }
    this.position = { x, y };
  }

  public closeAutoMove() {
    this.autoMove = false;
  }

  public openAutoMove() {
    this.randomTrajectory();
    this.autoMove = true;
  }

  public randomTrajectory() {
    const slopeBig = random(1, this.canvas.height / 2000);
    const slopeSmall = Math.random();
    const direction = Math.random() > 0.5 ? 1 : -1;
    const slope = (Math.random() > 0.5 ? slopeBig : slopeSmall) * direction;
    const offset = this.position.y - slope * this.position.x;
    this.slope = slope;
    this.offset = offset;
  }

  public initPosition() {
    this.position = {
      x: random(1, this.canvas.width - 1),
      y: random(1, this.canvas.height - 1),
    };
  }
}

export default Dot;
