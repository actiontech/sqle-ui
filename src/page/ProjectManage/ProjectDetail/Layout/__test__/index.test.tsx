import { SystemRole } from '../../../../../data/common';
import { projectDetailRouterConfig } from '../../../../../router/config';
import { generateNavigateMenu } from '../index';

describe('test ProjectManage/ProjectDetailLayout/index', () => {
  test('should match snapshot', () => {
    expect(
      generateNavigateMenu(projectDetailRouterConfig, SystemRole.admin, 'test')
    ).toMatchSnapshot();

    expect(
      generateNavigateMenu(projectDetailRouterConfig, '', 'test')
    ).toMatchSnapshot();
  });
});
