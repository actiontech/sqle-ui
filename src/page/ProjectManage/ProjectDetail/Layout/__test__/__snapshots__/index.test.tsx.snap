// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ProjectManage/ProjectDetailLayout/index should match snapshot 1`] = `
Array [
  Object {
    "icon": <ForwardRef(ProjectOutlined) />,
    "key": "projectOverview",
    "label": <CustomLink
      to="project/test/overview"
    >
      menu.projectOverview
    </CustomLink>,
    "title": "menu.projectOverview",
  },
  Object {
    "icon": <ForwardRef(ConsoleSqlOutlined) />,
    "key": "order",
    "label": <CustomLink
      to="project/test/order"
    >
      menu.order
    </CustomLink>,
    "title": "menu.order",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "icon": undefined,
            "key": "auditPlan",
            "label": <CustomLink
              to="project/test/auditPlan"
            >
              menu.auditPlane
            </CustomLink>,
            "title": "menu.auditPlane",
          },
        ],
        "key": "",
        "label": "",
        "type": "group",
      },
    ],
    "icon": <ForwardRef(CiCircleOutlined) />,
    "key": "plane",
    "label": "menu.auditPlane",
  },
  Object {
    "icon": <ForwardRef(DatabaseOutlined) />,
    "key": "data",
    "label": <CustomLink
      to="project/test/data"
    >
      menu.dataSource
    </CustomLink>,
    "title": "menu.dataSource",
  },
  Object {
    "icon": <ForwardRef(UserOutlined) />,
    "key": "member",
    "label": <CustomLink
      to="project/test/member"
    >
      menu.member
    </CustomLink>,
    "title": "menu.member",
  },
  Object {
    "icon": <ForwardRef(AuditOutlined) />,
    "key": "ruleTemplate",
    "label": <CustomLink
      to="project/test/rule/template"
    >
      menu.ruleTemplate
    </CustomLink>,
    "title": "menu.ruleTemplate",
  },
  Object {
    "icon": <ForwardRef(NodeIndexOutlined) />,
    "key": "progress",
    "label": <CustomLink
      to="project/test/progress"
    >
      menu.progressManage
    </CustomLink>,
    "title": "menu.progressManage",
  },
  Object {
    "icon": <ForwardRef(ProfileOutlined) />,
    "key": "Whitelist",
    "label": <CustomLink
      to="project/test/whitelist"
    >
      menu.whitelist
    </CustomLink>,
    "title": "menu.whitelist",
  },
]
`;

exports[`test ProjectManage/ProjectDetailLayout/index should match snapshot 2`] = `
Array [
  Object {
    "icon": <ForwardRef(ProjectOutlined) />,
    "key": "projectOverview",
    "label": <CustomLink
      to="project/test/overview"
    >
      menu.projectOverview
    </CustomLink>,
    "title": "menu.projectOverview",
  },
  Object {
    "icon": <ForwardRef(ConsoleSqlOutlined) />,
    "key": "order",
    "label": <CustomLink
      to="project/test/order"
    >
      menu.order
    </CustomLink>,
    "title": "menu.order",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "icon": undefined,
            "key": "auditPlan",
            "label": <CustomLink
              to="project/test/auditPlan"
            >
              menu.auditPlane
            </CustomLink>,
            "title": "menu.auditPlane",
          },
        ],
        "key": "",
        "label": "",
        "type": "group",
      },
    ],
    "icon": <ForwardRef(CiCircleOutlined) />,
    "key": "plane",
    "label": "menu.auditPlane",
  },
  Object {
    "icon": <ForwardRef(DatabaseOutlined) />,
    "key": "data",
    "label": <CustomLink
      to="project/test/data"
    >
      menu.dataSource
    </CustomLink>,
    "title": "menu.dataSource",
  },
  Object {
    "icon": <ForwardRef(UserOutlined) />,
    "key": "member",
    "label": <CustomLink
      to="project/test/member"
    >
      menu.member
    </CustomLink>,
    "title": "menu.member",
  },
  Object {
    "icon": <ForwardRef(AuditOutlined) />,
    "key": "ruleTemplate",
    "label": <CustomLink
      to="project/test/rule/template"
    >
      menu.ruleTemplate
    </CustomLink>,
    "title": "menu.ruleTemplate",
  },
  Object {
    "icon": <ForwardRef(NodeIndexOutlined) />,
    "key": "progress",
    "label": <CustomLink
      to="project/test/progress"
    >
      menu.progressManage
    </CustomLink>,
    "title": "menu.progressManage",
  },
  Object {
    "icon": <ForwardRef(ProfileOutlined) />,
    "key": "Whitelist",
    "label": <CustomLink
      to="project/test/whitelist"
    >
      menu.whitelist
    </CustomLink>,
    "title": "menu.whitelist",
  },
]
`;
